### ISSUE TITLE
`except`-target causes spurious `no-member`

### ISSUE THREAD

**Issue by peniblec** [PyCQA/pylint#2856]
### Steps to reproduce

1. Install GitPython: `pip3 install --user GitPython` (version on my
   setup: 2.1.11)

2. Add the following to `repro.py`:

``` python
"""Reproduce spurious linter warning."""

import git


try:
    print(git.Repo().git.status())
except git.exc.InvalidGitRepositoryError:
    print('Where am I?')
```

3. Run pylint: `pylint repro.py`

### Current behavior

    ************* Module repro
    linterfail.py:8:7: E1101: Instance of 'Exception' has no 'InvalidGitRepositoryError' member (no-member)
    linterfail.py:8:7: E1101: Instance of 'GitError' has no 'InvalidGitRepositoryError' member (no-member)

### Expected behavior

No warning.

### pylint --version output

    pylint 2.3.1
    astroid 2.2.5
    Python 3.5.3 (default, Sep 27 2018, 17:25:39) 
    [GCC 6.3.0 20170516]

### Comments

While there are many E1101-related issues open already, none of them
have the same cause AFAICT.  In this case, I think the warnings come
from these snippets in `git/__init__.py`:

    except GitError as exc:
    except Exception as exc:

I.e. these `except` blocks introduce the variable `exc`, whose name
conflicts with the `git.exc` module.  Renaming these variables makes
pylint's warnings disappear.

There are some obvious workarounds:

1. Since `git/__init__.py` does `from git.exc import *`, `repro.py`
   can just say `except git.GitRepositoryError`.

2. The GitPython team could simply rename these variables.

However, if I read the Python docs correctly, technically pylint
should not get confused by these variables, as targets of `except`
clauses do not outlive the clause.  From `help('try')`:

> When an exception has been assigned using "as target", it is cleared
> at the end of the except clause.  This is as if
>
>     except E as N:
>         foo
>
> was translated to
>
>     except E as N:
>         try:
>             foo
>         finally:
>             del N

I am raising this issue in case the fix to pylint is not too complex,
but I would understand if it turned out to be too complicated.

**Comment by PCManticore** [comment]
Thanks for opening an issue @peniblec It's possible that right now `pylint` might be confused about the exception bindings, but it's something we should definitely fix going forward.

**Comment by jacobtylerwalls** [comment]
#### minimal example without exception handlers

git/__ init __.py

```python
class Exc:
    pass
```

git/exc.py
```python
class MyClass
    pass
```

a.py
```python
"""python imports MyClass but pylint imports __init__.exc"""
import git.exc
print(git.exc.MyClass)

```

python a.py
```
<class 'git.exc.MyClass'>
```

pylint a.py
```
************* Module a
a.py:3:6: E1101: Class 'exc' has no 'MyClass' member (no-member)
```
